core/src/main/java/jenkins/tasks/SimpleBuildStep.java

```diff
@@ -104,15 +104,51 @@ default void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNu
      */
     default void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNull EnvVars env, @NonNull Launcher launcher,
                          @NonNull TaskListener listener) throws InterruptedException, IOException {
-        // If this is called, this must be an implementer of the previous API, in which case we call that, discarding
+        // If this does not require a workspace, defer to the version that does not take a workspace and launcher.
+        if (!this.requiresWorkspace()) {
+            this.perform(run, env, listener);
+            return;
+        }
+        // If we get here, this must be an implementer of the previous API, in which case we call that, discarding
         // the environment we were given.
         // But for that to work, that API method must have been implemented.
         if (Util.isOverridden(SimpleBuildStep.class, this.getClass(),
                 "perform", Run.class, FilePath.class, Launcher.class, TaskListener.class)) {
             this.perform(run, workspace, launcher, listener);
         } else {
-            throw new AbstractMethodError();
+            throw new AbstractMethodError(Messages.SimpleBuildStep_NeedPerformMethodWithWorkspace());
```

Comment:
This is for developers so should not be a localized message.

Reply:
Figured as much; if this path is taken, I'll delocalize them.